packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -190,17 +190,28 @@ export default class ConfigParser {
     }
 
     /**
-     * get excluded files from config pattern
+     * determine what specs to run based on the spec(s), suite(s), exclude
+     * attributes from CLI, config and capabilities
      */
-    getSpecs(capSpecs?: string[], capExclude?: string[]) {
-        let specs = ConfigParser.getFilePaths(this._config.specs!, undefined, this._pathService)
-        let spec = Array.isArray(this._config.spec) ? this._config.spec : []
+    getSpecs(capSpecs?: Spec[], capExclude?: Spec[]) {
+        const isSpecParamPassed = Array.isArray(this._config.spec)
+        // when CLI --spec is explicitly specified, this._config.specs contains the filtered
+        // specs matching the passed pattern else the specs defined inside the config are returned
+        let specs = ConfigParser.getFilePaths(this._config.specs, undefined, this._pathService)
         let exclude = ConfigParser.getFilePaths(this._config.exclude!, undefined, this._pathService)
         let suites = Array.isArray(this._config.suite) ? this._config.suite : []
 
-        /**
-         * check if user has specified a specific suites to run
-         */
+        // only use capability excludes if (CLI) --exclude or config exclude are not defined
```

Comment:
@dragosMC91 @christian-bromann Why is that? 
I have a case where I want to exclude tests via `config: { exclude` and via `capabilities: {
                ....,
                'wdio:exclude': [`
                 
Can we change that to not be the case and use both?
I changed the code to work for my case:
```
        if (Array.isArray(capExclude)) {
            exclude = [ ...exclude, ...ConfigParser.getFilePaths(capExclude, this._config.rootDir, this._pathService)]
        }
```
Is that acceptable?

Reply:
Yeah, I see 3 tests failing, seems this is a no go .. Anyway I'm curious why cannot both excludes work together and be merged?

Reply:
I'ld suggest to create a new issue and treat this as a feature request. Thanks!